schema/profile.js

```diff
@@ -300,10 +300,15 @@ class Profile extends BaseModel {
             .where({ establishmentId, role: 'admin' })
             .where('profiles.id', ref('permissions.profileId'));
 
-          return builder
+          builder
             .whereExists(role)
-            .orWhereExists(admin)
-            .orWhere('profiles.id', userId);
+            .orWhereExists(admin);
+
+          if (includeSelf) {
+            builder.orWhere('profiles.id', userId);
```

Comment:
It doesn't feel to me like this belongs in this method at all tbh. If we're making changes here then I'd remove the "or self" bit completely and have the function actually just do what it says it does, and add the additional `orWhere` as applicable at the point of implementation.

Reply:
I agree in principle, but this method doesn't return the query, it returns the executed results of several queries (results, total count, list of filters).

I could make a duplicate function e.g. `getNamedProfilesAndSelf()` but I'm not sure that's any better?

Reply:
It may not be fit for both of the purposes we're using it for then, since the total counts etc will need to be different depending on which context it is (i.e. filter applied or permissions restriction)

Reply:
Having said that, as long as the counts are correct in both contexts then this is fine.

Reply:
Yeah it passes the query built here to the count function - I'm hesitant to unpick it all as it feels like the whole lot could unravel.